### PR TITLE
Add GitHub based Flite voice

### LINKED ISSUE #111: Implement CMU g2p for Flite voice
For the Flite voice, we need the CMU Phonemeset to be implemented

### PR DESCRIPTION
**Pull request by lumpidu**
## Add Flite unit selection voice driver integration for native C/C++ adaption

- Adapt TTSEngineFlite.java
- Add modified logging from G2P native module and de-/initialize it in every JNI call (this needs to be improved)
- Use dlopen()/dlsym()/dlcose() for manual voice linkage

Limitation: cannot unload a loaded voice currently.

## Add RemoteAssets to make remote voices available

- introduce new Pojo's for adapting voices to the release asset format
- Implement remoteasset.VoiceRepo() to access GitHub release assets
- add a few new helper functions to FileUtils.java
- add unit tests

## Implement Downloading of Voice and integrate Flite unit selection voice

This implements downloading of the voices by adding a (deprecated)
ProgressDialog into VoiceInfo. One can see the progress while the file is
downloading. Still quite improvable, though, especially because the
chosen widget is marked as deprecated. It's possible to cancel an ongoing
download. An already downloaded voice can currently only be deleted by
cleaning the application data from the Android settings.

- Add classes AsyncThread for creating yet another async worker
- Add class Decompress for decompressing ZIP files
- augment pojos DeviceCoice, DeviceVoiceFiles and DeviceVoices
- add new boolean parameter fliterUnknown for white listing only known
  phonemes
- add a new TTSAudioControl for playing 16kHz audio
- sort order of the voice list is determined by voice name instead of type

## G2P

- fix memory leak when deallocating
- remove disturbing logging

## Move FileUtils
- Move FileUtils => com.grammatek.simaromur.utils

##  Remove all dependencies to the previous flite library

We have fully removed all code from original Flite projects and therefore can also remove the LICENSE-CMU.txt.
Although the Flite voice is based on the original Flite framework, Símarómur itself is not anymore depending on it.

implements / fixes #111 
implements / fixes #112

### CODE REVIEW

File: app/src/main/java/com/grammatek/simaromur/VoiceInfo.java [hunk 1]
```diff
@@ -184,6 +293,24 @@ public void onSpinnerClicked(View v) {
         toggleSpeakButton();
     }
 
+    public void onDownloadClicked(View v) {
+        Log.v(LOG_TAG, "onDownloadClicked");
+        toggleDownloadButton();
+        mProgressDialog = new ProgressDialog(this);
+        mProgressDialog.setMax(100); // Progress Dialog Max Value
```
Comment (lumpidu) [inline app/src/main/java/com/grammatek/simaromur/VoiceInfo.java:300]: Todo: remove comments

File: app/src/main/java/com/grammatek/simaromur/VoiceListAdapter.java [hunk 2]
```diff
@@ -48,8 +48,10 @@ public void onBindViewHolder(VoiceViewHolder holder, int position) {
         holder.voiceNameItemView.setText(current.name);
         if (current.needsNetwork()) {
             holder.voiceTypeItemView.setText(App.getContext().getResources().getString(R.string.type_network));
+        } else if (current.needsDownload()) {
+                holder.voiceTypeItemView.setText(App.getContext().getResources().getString(R.string.type_local_downloaded));
         } else {
-            holder.voiceTypeItemView.setText(App.getContext().getResources().getString(R.string.type_local));
+                holder.voiceTypeItemView.setText(App.getContext().getResources().getString(R.string.type_local));
```
Comment (lumpidu) [inline app/src/main/java/com/grammatek/simaromur/VoiceListAdapter.java:54]: indentation

File: app/src/main/java/com/grammatek/simaromur/db/VoiceDao.java [hunk 3]
```diff
@@ -53,6 +51,10 @@ Voice findVoice(String name, String internalName, String languageCode, String la
     @Query("SELECT * FROM voice_table WHERE url in ('assets') ")
     List<Voice> getAssetVoices();
 
+    // Return voices registered in Assets
```
Comment (lumpidu) [inline app/src/main/java/com/grammatek/simaromur/db/VoiceDao.java:54]: misleading comment

File: app/src/main/java/com/grammatek/simaromur/device/DownloadVoiceManager.java [hunk 4]
```diff
@@ -0,0 +1,578 @@
...
+    private boolean cacheVoiceDescriptionToDisk(String filePath, DeviceVoices voices) {
+        // cache the voice description from server
+        Gson voiceDescriptionGson = new GsonBuilder().create();
+        String voiceDescriptionJson = voiceDescriptionGson.toJson(voices);
+        return FileUtils.writeFileToStorage(filePath,
+                voiceDescriptionJson.getBytes(StandardCharsets.UTF_8));
+    }
+
+    /**
+     * Returns the combines voice information from disk and repository, if available
```
Comment (lumpidu) [inline app/src/main/java/com/grammatek/simaromur/device/DownloadVoiceManager.java:251]: typo

**Review by bnika — APPROVED** [review]

**Pull request merged**